packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -40,12 +40,18 @@ export function extractLogSafeErrorProperties(error: any, sanitizeStack: boolean
         ? error.message as string
         : String(error);
 
-    const safeProps: { message: string; errorType?: string; stack?: string } = {
+    const safeProps: {
+        message: string;
+        errorType?: string;
+        stack?: string;
+        canRetry?: boolean,
+        retryAfterSeconds?: number,
```

Comment:
This seems wrong. Why this code needs to know and special-case these specific properties? Why it's not done through more generic getTelemetryProperties() mechanism?
I think this code should not care about specific structure of driver errors.

Reply:
Oh, it's wrapping, not logging. I'd still say this code should not know anything about driver specifics

Reply:
> I think this code should not care about specific structure of driver errors.

Yeah, I see your point when you put it that way.

I'm bothered by the fact that we fiddle with these non-typesafe props, so I was more willing to hack around to support them.

Using telemetry props is fine and will work.  But also doesn't sit perfectly with me, considering for several months I was lobbying for telemetry props to be divorced from the props of the object itself.

For now I will try using telemetry props and see if any concerns come to mind.

Reply:
Well, should normalization always preserve both object properties and getTelemetryProperties()? If one of the purposes / side-effects of normalization is to clone object, then maybe we need to add clone() method on interface to let it preserve all of the semantics of the object? Feels like it comes back to user cans discussion, i.e., what are the expectations developers have (or must have) around similar interactions?
But I'd try really hard not to special case properties like that - there is just no end in sight for this process as people will find an opportunity to add more.

The other side of the coin (what code was doing before) is that all the wrapping is only for logging only. Action is done on original object only. That works in simple cases (like this), but it's hard to make it happen in general case (as layers quite often want to annotate object before rethrowing)

Reply:
So... if the error already has errorType and getTelemetryProperties (which always go together... should be enforced in base types but I think it's not quite), then normalizeError will return the same error.  My fear is that someone may shove `canRetry` on some arbitrary object.  If everyone set `canRetry` via `addTelemetryProperties` then this won't be necessary.

I'll ponder and update the PR tomorrow I guess.

Reply:
Great example of why I am eager for `catch` variables to default to `unknown`, so we will not treat them as `any` and slap random stuff on them.

Reply:
I opened https://github.com/microsoft/FluidFramework/issues/8743 about this, and for now I'm copying over those props specifically in this codepath.